kubectl-plugin/pkg/cmd/log/log_test.go

```diff
@@ -180,107 +183,71 @@ func TestRayClusterLogValidate(t *testing.T) {
 		{
 			name: "Test validation when no context is set",
 			opts: &ClusterLogOptions{
-				configFlags: genericclioptions.NewConfigFlags(false),
-				outputDir:   fakeDir,
-				args:        []string{"fake-cluster"},
-				nodeType:    "head",
-				ioStreams:   &testStreams,
+				configFlags:  genericclioptions.NewConfigFlags(false),
+				outputDir:    fakeDir,
+				ResourceName: "fake-cluster",
+				nodeType:     "head",
+				ioStreams:    &testStreams,
 			},
 			expectError: "no context is currently set, use \"kubectl config use-context <context>\" to select a new one",
 		},
-		{
-			name: "Test validation when more than 1 arg",
-			opts: &ClusterLogOptions{
-				// Use fake config to bypass the config flag checks
-				configFlags: fakeConfigFlags,
-				outputDir:   fakeDir,
-				args:        []string{"fake-cluster", "another-fake"},
-				nodeType:    "head",
-				ioStreams:   &testStreams,
-			},
-			expectError: "must have at only one argument",
-		},
-		{
-			name: "Test validation when node type is `all`",
-			opts: &ClusterLogOptions{
-				// Use fake config to bypass the config flag checks
-				configFlags: fakeConfigFlags,
-				outputDir:   fakeDir,
-				args:        []string{"fake-cluster"},
-				nodeType:    "all",
-				ioStreams:   &testStreams,
-			},
-			expectError: "node type `all` is currently not supported",
-		},
-		{
-			name: "Test validation when node type is `worker`",
-			opts: &ClusterLogOptions{
-				// Use fake config to bypass the config flag checks
-				configFlags: fakeConfigFlags,
-				outputDir:   fakeDir,
-				args:        []string{"fake-cluster"},
-				nodeType:    "worker",
-				ioStreams:   &testStreams,
-			},
-			expectError: "node type `worker` is currently not supported",
-		},
```

Comment:
Shouldn't we also add tests for these 2 nodeType?

Reply:
I'm not sure what the value add would be for testing the nodeType here since the code path is the same as `head` with the exception of what goes into the `ListOption` field which is in the `Validate` func. I think adding e2e tests for would add more value to it. 

Will work on adding e2e tests for `log` and `submit` after https://github.com/ray-project/kuberay/issues/2419

Reply:
e2e tests sounds good!